carbon-design-system/carbon
# WIP feat(tabs): add experimental tabs component

Closes #1293 
Adds experimental tabs component.

#### Changelog

**New**
- adds  experimental tabs styling to `_tabs.scss`
- adds experimental tab styling variables to `_theme-tokens.scss` 
- adds new class `.#{$prefix}--tabs-trigger--open` for arrow motion on open/close in dropdown view
- adds testing for disabled tab and new `.#{$prefix}--tabs-trigger--open` class
- adds new `.#{$prefix}--tabs-trigger--open` class to default tab options in `tabs.js`
- adds functionality to include `disabled` in html `li` element if tab is disabled
- adds `@mixin` to `_layout.scss` to use for mobile media queries (allows for *max-width*, in this case used particularly for styling the tab dropdown in mobile view)
- adds `:active` state on links for Safari focus states

**Changed**
- refactor `_tabs.scss` file to use `@mixin`
- refactor scss code for experimental tabs to no longer be nested (except for `@include breakpoint()`'s.
- refactor `tabs.config.js` to include a disabled tab item
- refactor `tabs.js` to include functionality for disabled tab
- refactor `_updateMenuState(force)` to toggle new `.#{$prefix}--tabs-trigger--open`

**Testing/Reviewing**
- Ensure *disabled* styling is functioning correctly on hover, focus, select on click and keyboard navigation.
- Ensure mobile view is functioning correctly.
- Ensure focus states in Safari are working.
- Staging link: http://carbon-dev-environment-tabs-x-grateful-camel.mybluemix.net/component/tabs--default.html
